`tests/device_tests/ethereum/test_signtx.py::test_signtx_payment_req` OOMs on T2B1

Some of the above test variants fail with an OOM.

**Describe the bug**
```
[13:46:50.686] 44.312 trezor.wire.message_handler INF [USBIF] received message: EthereumSignTx
[13:46:50.723] 44.349 trezor.loop DBG spawn new task: <generator object 'with_context' at 0x2001de00>
[13:46:50.724] 44.349 trezor.workflow DBG start: <generator object 'with_context' at 0x2001de00>
[13:46:51.178] 44.804 trezor.ui DBG UI task was stopped: <generator object '_handle_button_events' at 0x2001b800>
[13:46:51.180] 44.805 trezor.ui DBG UI task was stopped: <generator object 'usb_checker_task' at 0x2001b8b0>
[13:46:51.216] 44.842 trezor.loop DBG close spawned task: <generator object 'homescreen' at 0x2001a580>
[13:46:51.218] 44.843 trezor.workflow DBG default closed: <generator object 'homescreen' at 0x2001a580>
[13:46:51.224] 44.850 trezor.wire.codec.codec_context DBG [USBIF] expect: None
[13:46:51.225] 44.850 trezor.loop DBG finish: <generator object '__iter__' at 0x2001f2e0>
[13:46:51.226] 44.851 trezor.wire.protocol_common INF ButtonRequest sent: confirm_payment_request
[13:46:51.228] 44.851 trezor.wire.codec.codec_context DBG [USBIF] write: ButtonRequest
[13:46:51.230] 44.853 trezor.wire.codec.codec_context DBG [USBIF] expect: ButtonAck
[13:46:51.231] 44.855 trezor.wire.codec.codec_context DBG [USBIF] read: ButtonAck
[13:46:51.233] 44.855 trezor.wire.protocol_common INF ButtonRequest acked: confirm_payment_request
[13:46:51.234] 44.856 trezor.loop DBG finish: <generator object '__iter__' at 0x2001af80>
[13:46:51.236] 44.858 apps.debug DBG 1 receive: <DebugLinkGetState>
[13:46:51.244] 44.870 trezor.wire.codec.codec_context DBG [USBIF] write: DebugLinkState
[13:46:51.247] 44.872 trezor.wire.codec.codec_context DBG [USBIF] expect: None
[13:46:51.259] 44.885 trezor.loop DBG finish: <generator object '__iter__' at 0x20023000>
[13:46:51.260] 44.886 apps.debug DBG 1 receive: <DebugLinkDecision>
[13:46:51.263] 44.889 trezor.ui DBG UI task was stopped: <generator object '_handle_button_events' at 0x2001ef30>
[13:46:51.266] 44.891 trezor.loop DBG finish: <generator object '__iter__' at 0x20023720>
[13:46:51.267] 44.893 apps.debug DBG 1 receive: <DebugLinkGetState>
[13:46:51.268] 44.894 trezor.wire.codec.codec_context DBG [USBIF] write: DebugLinkState
[13:46:51.270] 44.896 trezor.loop DBG finish: <generator object '__iter__' at 0x2001fbd0>
[13:46:51.271] 44.897 trezor.loop DBG finish: <generator object 'br_task' at 0x2001f060>
[13:46:51.273] 44.897 trezor.ui DBG UI task exited by itself: <generator object 'br_task' at 0x2001f060>
[13:46:51.274] 44.897 trezor.loop DBG finish: <generator object '__iter__' at 0x20023910>
[13:46:51.279] 44.905 trezor.loop DBG finish: <generator object '__iter__' at 0x20024060>
[13:46:51.280] 44.906 apps.debug DBG 1 receive: <DebugLinkGetState>
[13:46:51.292] 44.918 trezor.wire.codec.codec_context DBG [USBIF] write: DebugLinkState
[13:46:51.297] 44.923 trezor.wire.codec.codec_context DBG [USBIF] expect: None
[13:46:51.310] 44.936 trezor.loop DBG finish: <generator object '__iter__' at 0x2002a730>
[13:46:51.311] 44.937 apps.debug DBG 1 receive: <DebugLinkDecision>
[13:46:51.326] 44.952 trezor.ui DBG UI task was stopped: <generator object '_handle_button_events' at 0x20024760>
[13:46:51.328] 44.954 trezor.loop DBG finish: <generator object '__iter__' at 0x2002b750>
[13:46:51.330] 44.955 apps.debug DBG 1 receive: <DebugLinkGetState>
[13:46:51.331] 44.957 trezor.wire.codec.codec_context DBG [USBIF] write: DebugLinkState
[13:46:51.333] 44.958 trezor.loop DBG finish: <generator object '__iter__' at 0x20029730>
[13:46:51.334] 44.959 trezor.loop DBG finish: <generator object 'br_task' at 0x20024840>
[13:46:51.335] 44.959 trezor.ui DBG UI task exited by itself: <generator object 'br_task' at 0x20024840>
[13:46:51.337] 44.960 trezor.loop DBG finish: <generator object '__iter__' at 0x2002b8f0>
[13:46:51.353] 44.978 trezor.loop DBG finish: <generator object '__iter__' at 0x2002ce20>
[13:46:51.354] 44.980 apps.debug DBG 1 receive: <DebugLinkGetState>
[13:46:51.367] 44.993 trezor.wire.codec.codec_context DBG [USBIF] write: DebugLinkState
[13:46:51.372] 44.998 trezor.wire.codec.codec_context DBG [USBIF] expect: None
[13:46:51.386] 45.012 trezor.loop DBG finish: <generator object '__iter__' at 0x2001c070>
[13:46:51.387] 45.013 apps.debug DBG 1 receive: <DebugLinkGetState>
[13:46:51.400] 45.026 trezor.wire.codec.codec_context DBG [USBIF] write: DebugLinkState
[13:46:51.418] 45.044 trezor.loop DBG finish: <generator object '__iter__' at 0x20022820>
[13:46:51.419] 45.045 apps.debug DBG 1 receive: <DebugLinkDecision>
[13:46:51.421] 45.047 trezor.ui DBG UI task was stopped: <generator object '_handle_button_events' at 0x20010410>
[13:46:51.423] 45.049 trezor.loop DBG finish: <generator object '__iter__' at 0x20022f00>
[13:46:51.425] 45.051 apps.debug DBG 1 receive: <DebugLinkGetState>
[13:46:51.426] 45.052 trezor.wire.codec.codec_context DBG [USBIF] write: DebugLinkState
[13:46:51.428] 45.054 trezor.loop DBG finish: <generator object '__iter__' at 0x2001b690>
[13:46:51.429] 45.054 trezor.loop DBG finish: <generator object 'br_task' at 0x20014040>
[13:46:51.430] 45.055 trezor.ui DBG UI task exited by itself: <generator object 'br_task' at 0x20014040>
[13:46:51.432] 45.055 trezor.loop DBG finish: <generator object '__iter__' at 0x200230e0>
[13:46:51.436] 45.062 trezor.loop DBG finish: <generator object '__iter__' at 0x20023860>
[13:46:51.438] 45.064 apps.debug DBG 1 receive: <DebugLinkDecision>
[13:46:51.453] 45.078 trezor.loop DBG finish: <generator object '__iter__' at 0x20024110>
[13:46:51.454] 45.080 apps.debug DBG 1 receive: <DebugLinkGetState>
[13:46:51.455] 45.081 trezor.wire.codec.codec_context DBG [USBIF] write: DebugLinkState
[13:46:51.457] 45.083 trezor.wire.codec.codec_context DBG [USBIF] expect: None
[13:46:51.458] 45.084 trezor.loop DBG finish: <generator object '__iter__' at 0x20024ed0>
[13:46:51.460] 45.086 apps.debug DBG 1 receive: <DebugLinkDecision>
[13:46:51.474] 45.100 trezor.ui DBG UI task was stopped: <generator object '_handle_button_events' at 0x2001f8c0>
[13:46:51.477] 45.102 trezor.loop DBG finish: <generator object '__iter__' at 0x20027260>
[13:46:51.478] 45.104 apps.debug DBG 1 receive: <DebugLinkGetState>
[13:46:51.479] 45.105 trezor.wire.codec.codec_context DBG [USBIF] write: DebugLinkState
[13:46:51.481] 45.107 trezor.loop DBG finish: <generator object '__iter__' at 0x20024450>
[13:46:51.482] 45.108 trezor.loop DBG finish: <generator object 'br_task' at 0x2001f9a0>
[13:46:51.484] 45.108 trezor.ui DBG UI task exited by itself: <generator object 'br_task' at 0x2001f9a0>
[13:46:51.485] 45.108 trezor.loop DBG finish: <generator object '__iter__' at 0x20027450>
[13:46:51.497] GC: total: 163392, used: 115056, free: 48336
[13:46:51.497]  No. of 1-blocks: 2540, 2-blocks: 495, max blk sz: 512, max free sz: 98
[13:46:51.498] 45.123 trezor.loop ERR exception:
[13:46:51.499] Traceback (most recent call last):
[13:46:51.500]   File "trezor/loop.py", in _step
[13:46:51.500]   File "trezor/wire/context.py", in with_context
[13:46:51.501]   File "apps/ethereum/keychain.py", in wrapper
[13:46:51.502]   File "apps/ethereum/sign_tx.py", in sign_tx
[13:46:51.503]   File "apps/ethereum/sign_tx.py", in confirm_data_and_summary
[13:46:51.504]   File "apps/ethereum/layout.py", in require_confirm_payment_request
[13:46:51.505]   File "trezor/ui/layouts/caesar/__init__.py", in confirm_payment_request
[13:46:51.506]   File "trezor/ui/layouts/menu.py", in confirm_with_menu
[13:46:51.507]   File "trezor/ui/layouts/menu.py", in interact_with_menu
[13:46:51.508]   File "trezor/ui/layouts/menu.py", in show_menu
[13:46:51.509]   File "trezor/ui/layouts/menu.py", in _interact
[13:46:51.509]   File "trezor/ui/layouts/caesar/__init__.py", in <lambda>
[13:46:51.510] MemoryError: 
```

**Firmware version and revision**
Reproduced on `main` (5e97e32f7f) and with context-based UI layouts (3df5ac4124).

**To Reproduce**
```
XTASK_BUILD_OPTS="--dbg-console=vcp" STORAGE_INSECURE_TESTING_MODE=1 PYOPT=0 TREZOR_MODEL=T2B1 BITCOIN_ONLY=0 QUIET_MODE=1 make -C core clean build_firmware upload
pytest -v tests/device_tests/ethereum/test_signtx.py::test_signtx_payment_req
```

Happens also on the old build system.

Host-side logs:
```
BYTES    trezorlib.protocol_v1:protocol_v1.py:323 encoded as type 58 (562 bytes): 08ac8080800808bc808080080880808080080800080012001a0504a817c800220252085a2a307838654137613366636343323131454434386237363362343136343838344444626346336230413938320502540be4003a004000480162680a6674727a643100ffffffff18000801120446414b45183c220c46616b65206e6574776f726b0007ec9f9164ad0b80432ecd7152c62f2c0df4161be08c65e4db556d6d1a146c7c44d19468b7c3c66e638c0d5d71b0cf985d0dedc064ec881e02257a52e17f22c103680072e7020a200332bb24623eb452af0a981adff2d8ee9c3b50f42f6545aad5ef9e248933729712097472657a6f722e696f1a6c1a6a0800120a302e30363336204254431a22314a4164375843427a47756447704a5153445366706d4a68697967744c515761474c20ac80808008208080808008208080808008200020002a20668ac47462b3208d934b775ae0c25d404923e29baef783f764e971be4688a5051a6612640a2a30783733643033383546346438453030433565363530344336303330463437424636323132373336413810ac8080800810bc80808008108080808008100010001a20e541a033524647b3420c0c3d4178d8a65eadff5efb33479911aa2c0adcc046ab322000e40b54020000000000000000000000000000000000000000000000000000002a4040bc842cd90cedf9d7aa1ef95485443bb38f25291023fa8c4226c9323eee5f4365b09cbbb512af3f660e471a8235d5788c04745e1e44f045b370c3d8e0589ac1
BYTES    trezorlib.protocol_v1:protocol_v1.py:336 received type 26 (29 bytes): 080110012217636f6e6669726d5f7061796d656e745f72657175657374
DEBUG    trezorlib.protocol_v1:protocol_v1.py:342 received message: ButtonRequest
DEBUG    trezorlib.protocol_v1:protocol_v1.py:318 sending message: ButtonAck
BYTES    trezorlib.protocol_v1:protocol_v1.py:323 encoded as type 27 (0 bytes): 
BYTES    trezorlib.debuglink:debuglink.py:614 encoded as type 101 (4 bytes): 18022000
BYTES    trezorlib.debuglink:debuglink.py:614 encoded as type 100 (2 bytes): 0802
BYTES    trezorlib.debuglink:debuglink.py:614 encoded as type 101 (4 bytes): 18002001
BYTES    trezorlib.debuglink:debuglink.py:614 encoded as type 101 (4 bytes): 18022000
BYTES    trezorlib.debuglink:debuglink.py:614 encoded as type 100 (2 bytes): 4001
BYTES    trezorlib.debuglink:debuglink.py:614 encoded as type 101 (4 bytes): 18002001
BYTES    trezorlib.debuglink:debuglink.py:614 encoded as type 101 (4 bytes): 18022000
BYTES    trezorlib.debuglink:debuglink.py:614 encoded as type 101 (4 bytes): 18022000
BYTES    trezorlib.debuglink:debuglink.py:614 encoded as type 100 (2 bytes): 0801
BYTES    trezorlib.debuglink:debuglink.py:614 encoded as type 101 (4 bytes): 18002001
BYTES    trezorlib.debuglink:debuglink.py:614 encoded as type 100 (2 bytes): 4002
BYTES    trezorlib.debuglink:debuglink.py:614 encoded as type 101 (4 bytes): 18002001
BYTES    trezorlib.debuglink:debuglink.py:614 encoded as type 100 (2 bytes): 4001
BYTES    trezorlib.debuglink:debuglink.py:614 encoded as type 101 (4 bytes): 18002001
BYTES    trezorlib.debuglink:debuglink.py:614 encoded as type 101 (4 bytes): 18022000
BYTES    trezorlib.debuglink:debuglink.py:614 encoded as type 101 (4 bytes): 18022000
BYTES    trezorlib.debuglink:debuglink.py:614 encoded as type 100 (2 bytes): 0801
BYTES    trezorlib.debuglink:debuglink.py:614 encoded as type 101 (4 bytes): 18002001
BYTES    trezorlib.debuglink:debuglink.py:614 encoded as type 100 (2 bytes): 4002
BYTES    trezorlib.debuglink:debuglink.py:614 encoded as type 101 (4 bytes): 18002001
BYTES    trezorlib.debuglink:debuglink.py:614 encoded as type 100 (2 bytes): 0801
BYTES    trezorlib.debuglink:debuglink.py:614 encoded as type 101 (4 bytes): 18002001
BYTES    trezorlib.debuglink:debuglink.py:614 encoded as type 100 (2 bytes): 0801
BYTES    trezorlib.debuglink:debuglink.py:614 encoded as type 101 (4 bytes): 18002001
BYTES    trezorlib.protocol_v1:protocol_v1.py:336 received type 3 (4 bytes): 08631200
DEBUG    trezorlib.protocol_v1:protocol_v1.py:342 received message: Failure
------------------------------------------------------------------ Captured log teardown ------------------------------------------------------------------
BYTES    trezorlib.debuglink:debuglink.py:614 encoded as type 9009 (0 bytes): 
DEBUG    trezorlib.debuglink:debuglink.py:968 GC info: free=89.12% max_free=88.71%

================================================================= short test summary info =================================================================
FAILED tests/device_tests/ethereum/test_signtx.py::test_signtx_payment_req[True-False-False] - trezorlib.exceptions.TrezorFailure: FirmwareError:
```

--- a/core/src/trezor/ui/layouts/caesar/__init__.py
+++ b/core/src/trezor/ui/layouts/caesar/__init__.py
@@ -577,42 +577,46 @@ async def confirm_payment_request(
         ) as obj:
             await raise_if_not_confirmed(obj, "confirm_payment_request")
 
-    menu_items = []
-    if recipient_address is not None:
-        menu_items.append(
-            create_details(TR.address__title_provider_address, recipient_address)
-        )
-    for refund in refunds:
-        refund_account_items: list[StrPropertyType] = [("", refund.address, None)]
-        if refund.account:
-            refund_account_items.append((TR.words__account, refund.account, None))
-        if refund.account_path:
-            refund_account_items.append(
-                (TR.address_details__derivation_path, refund.account_path, None)
+    async def _task() -> None:
+        menu_items = []
+        if recipient_address is not None:
+            menu_items.append(
+                create_details(TR.address__title_provider_address, recipient_address)
             )
-        menu_items.append(
-            create_details(
-                TR.address__title_refund_address,
-                refund_account_items,
+        for refund in refunds:
+            refund_account_items: list[StrPropertyType] = [("", refund.address, None)]
+            if refund.account:
+                refund_account_items.append((TR.words__account, refund.account, None))
+            if refund.account_path:
+                refund_account_items.append(
+                    (TR.address_details__derivation_path, refund.account_path, None)
+                )
+            menu_items.append(
+                create_details(
+                    TR.address__title_refund_address,
+                    refund_account_items,
+                )
             )
-        )
-    if menu_items:
-        menu = Menu.root(menu_items)
+        if menu_items:
+            menu = Menu.root(menu_items)
 
-        with trezorui_api.confirm_with_info(
-            title=title,
-            items=[(TR.words__provider, True), (recipient_name, False)],
-            verb=TR.buttons__continue,
-            verb_info=INFO_ICON,
-            external_menu=True,
-        ) as main_layout:
-            await confirm_with_menu(main_layout, menu, "confirm_payment_request")
-    else:
-        await confirm_properties(
-            "confirm_payment_request",
-            title,
-            [(TR.words__provider, recipient_name, True)],
-        )
+            with trezorui_api.confirm_with_info(
+                title=title,
+                items=[(TR.words__provider, True), (recipient_name, False)],
+                verb=TR.buttons__continue,
+                verb_info=INFO_ICON,
+                external_menu=True,
+            ) as main_layout:
+                await confirm_with_menu(main_layout, menu, "confirm_payment_request")
+        else:
+            await confirm_properties(
+                "confirm_payment_request",
+                title,
+                [(TR.words__provider, recipient_name, True)],
+            )
+
+    # Allow GC to free the objects allocated above.
+    await _task()
 
     for trade in trades:
         await confirm_trade(

